packages/react-storage/src/components/StorageBrowser/Views/Controls/Loading.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import type { OmitElements } from '../types';
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+
+const { Icon, Span, View } = StorageBrowserElements;
+
+const BLOCK_NAME = `${CLASS_BASE}__loading`;
+
+interface LoadingControlProps {
+  loadingText?: string | undefined;
+}
+
+export interface _LoadingControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends Pick<T, 'Icon' | 'Span' | 'View'> {
+  ({ loadingText }: LoadingControlProps): React.JSX.Element;
+}
+
+export interface LoadingControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends OmitElements<_LoadingControl<T>, 'Icon' | 'Text' | 'View'> {
+  ({ loadingText }: LoadingControlProps): React.JSX.Element;
+}
+
+export const LoadingControl: LoadingControl = ({ loadingText }) => (
+  <View className={BLOCK_NAME}>
+    <Icon className={`${BLOCK_NAME}__icon`} variant="loading" />
+    <Span className={`${BLOCK_NAME}__text`} aria-live="polite">
+      {loadingText ?? 'Loading'}
+    </Span>
```

Comment:
Is the text here required for ally, if so is it possible to use `aria-live` with hidden content? IMO there is not much context provided by a loading message that is not already conveyed in the controls and each additional piece of text we add requires further configuration properties

Reply:
Yes, we need some sort of context for screen readers so accessible text of some sort is needed. I have simplified this to just say "Loading" and not accept props in latest commit, because I don't think the custom text on each view provides that much more context. But, even if we accessibly hide this text, it will eventually need to be configurable for i18n.